pkg/handlers/archive.go

```diff
@@ -174,36 +174,35 @@ func (h *archiveHandler) openArchive(
 // to handle nested archives or to continue processing based on the file's content and depth in the archive structure.
 func (h *archiveHandler) extractorHandler(dataOrErrChan chan DataOrErr) func(context.Context, archives.FileInfo) error {
 	return func(ctx context.Context, file archives.FileInfo) error {
+		if common.IsDone(ctx) {
```

Comment:
I also bumped this up to the top; felt prudent to bail immediately if this is the case.